src/essence/Tools/Isochrone/IsochroneTool_Algorithm.js

```diff
@@ -0,0 +1,172 @@
+
+import Map_ from "../../Basics/Map_/Map_";
+const _X = 1, _Y = 0;
+
+//px contains array indices [y, x]
+const getPx = (arr, px) => arr[px[_Y]][px[_X]];
+const setPx = (arr, px, val) => arr[px[_Y]][px[_X]] = val;
+
+const h_getParentIndex = i => Math.ceil(i / 2) - 1;
+const h_getChildIndex = i => i * 2 + 1;
+
+class PxHeap {
```

Comment:
document purpose of class with a jsdoc-style comment

Reply:
was this the class you said you copied from somewhere? if so, please find it

Reply:
this is my own code, and now has documentation. I'm just a bit skeptical that it needs to be its own class instead of a set of functions, since it's only used in the function below, but given that it's there in part to make up for JS's lack of a native priority queue implementation, I feel ok about it.

Reply:
looks like you already made it a separate class. it's up to you! I can be convinced either way